packages/apps/plugins/plugin-table/src/components/TableMain.tsx

```diff
@@ -4,27 +4,35 @@
 
 import React, { type FC } from 'react';
 
-import { Main } from '@dxos/react-ui';
+import { DensityProvider, Main } from '@dxos/react-ui';
 import { Table } from '@dxos/react-ui-table';
 import { baseSurface } from '@dxos/react-ui-theme';
 
 import { ObjectTable, type ObjectTableProps } from './ObjectTable';
 
 const TableMain: FC<Omit<ObjectTableProps, 'getScrollElement'>> = ({ table }) => (
-  <Table.Root>
-    <Table.Viewport asChild>
-      <Main.Content
-        classNames={[baseSurface, 'fixed inset-inline-0 block-start-[--topbar-size] block-end-0 overflow-auto']}
-      >
-        <ObjectTable
-          table={table}
-          key={table.id} // New component instance per table.
-          stickyHeader
-          role='grid'
-        />
-      </Main.Content>
-    </Table.Viewport>
-  </Table.Root>
+  <Main.Content
+    classNames={[baseSurface, 'fixed inset-inline-0 block-start-[--topbar-size] block-end-0 overflow-hidden']}
+  >
+    <DensityProvider density='fine'>
+      <Table.Root>
+        <Table.Viewport asChild>
+          {/* TODO(burdon): Move into ObjectTable. */}
+          {/* TODO(burdon): Floating "add row" jumps by 1 pixel on scroll. */}
+          {/* TODO(burdon): Blue focus highlight is clipped by 1px on the left. */}
+          {/* TODO(burdon): Focused row and focus ring passes over the sticky header. */}
+          <div className='flex flex-col h-full overflow-auto'>
+            <ObjectTable
+              key={table.id} // New component instance per table.
+              table={table}
+              stickyHeader
+              role='grid'
+            />
+          </div>
+        </Table.Viewport>
```

Comment:
Instead of wrapping a primitive div we can omit the `asChild` prop.
```suggestion
        <Table.Viewport classNames="flex flex-col h-full overflow-auto">
          {/* TODO(burdon): Move into ObjectTable. */}
          {/* TODO(burdon): Floating "add row" jumps by 1 pixel on scroll. */}
          {/* TODO(burdon): Blue focus highlight is clipped by 1px on the left. */}
          {/* TODO(burdon): Focused row and focus ring passes over the sticky header. */}
            <ObjectTable
              key={table.id} // New component instance per table.
              table={table}
              stickyHeader
              role='grid'
            />
        </Table.Viewport>
```